evm/chains/sharding/mainchain_handler/mainchain_handler.py

```diff
@@ -0,0 +1,102 @@
+import time
+
+import rlp
+
+from eth_utils import (
+    to_checksum_address,
+)
+
+from evm.chains.sharding.mainchain_handler.config import (
+    GASPRICE,
+    TX_GAS,
+)
+
+
+class MainchainHandler:
+
+    def __init__(self, web3_instance, use_eth_tester=False):
```

Comment:
I'm also thinking that the `use_eth_tester` param should be removed.  I see that there are some conditional logic forks in place, however, I believe all of them are in methods that are specifically for testing purposes and I've proposed removing those from this object.

Reply:
No problem. Will handle the tester logics outside the class, in the tests, for now.